File: openbis_api/source/java/ch/ethz/sis/openbis/generic/dssapi/v3/dto/datasetfile/create/DataSetFileCreation.java
package ch.ethz.sis.openbis.generic.dssapi.v3.dto.datasetfile.create;

import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.create.ICreation;
import ch.systemsx.cisd.base.annotation.JsonObject;
import ch.systemsx.cisd.common.annotation.TechPreview;

@JsonObject("dss.dto.datasetfile.create.DataSetFileCreation")
@TechPreview
public class DataSetFileCreation implements ICreation
{

    private static final long serialVersionUID = 1L;

    private String path;

    private boolean directory;

    private Long fileLength;

    private Integer checksumCRC32;

    public String getPath()
    {
        return path;
    }

    public void setPath(String path)
    {
        this.path = path;
    }

    public boolean isDirectory()
    {
        return directory;
    }

    public void setDirectory(boolean directory)
    {
        this.directory = directory;
    }

    public Long getFileLength()
    {
        return fileLength;
    }

    public void setFileLength(Long fileLength)
    {
        this.fileLength = fileLength;
    }

    public Integer getChecksumCRC32()
    {
        return checksumCRC32;
    }

    public void setChecksumCRC32(Integer checksumCRC32)
    {
        this.checksumCRC32 = checksumCRC32;
    }
}

File: openbis/source/java/ch/systemsx/cisd/openbis/public/resources/api/v3/dss/dto/datasetfile/create/DataSetFileCreation.js
define([ "stjs" ], function(stjs) {
	var DataSetFileCreation = function() {
	};
	stjs.extend(DataSetFileCreation, null, [], function(constructor, prototype) {
		prototype['@type'] = 'dss.dto.datasetfile.create.DataSetFileCreation';
		constructor.serialVersionUID = 1;
		prototype.path = null;
		prototype.directory = null;
		prototype.fileLength = null;
		prototype.checksumCRC32 = null;
		prototype.checksum = null;
		prototype.checksumType = null;
		
		prototype.getPath = function() {
			return this.path;
		};
		prototype.setPath = function(path) {
			this.path = path;
		};
		prototype.isDirectory = function() {
			return this.directory;
		};
		prototype.setDirectory = function(directory) {
			this.directory = directory;
		};
		prototype.getFileLength = function() {
			return this.fileLength;
		};
		prototype.setFileLength = function(fileLength) {
			this.fileLength = fileLength;
		};
		prototype.getChecksumCRC32 = function() {
			return this.checksumCRC32;
		};
		prototype.setChecksumCRC32 = function(checksumCRC32) {
			this.checksumCRC32 = checksumCRC32;
		};
		prototype.getChecksum = function() {
			return this.checksum;
		};
		prototype.setChecksum = function(checksum) {
			this.checksum = checksum;
		};
		prototype.getChecksumType = function() {
			return this.checksumType;
		};
		prototype.setChecksumType = function(checksumType) {
			this.checksumType = checksumType;
		};
	}, {
	});
	return DataSetFileCreation;
})